sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -642,37 +632,73 @@ object PushFoldableIntoBranches extends Rule[LogicalPlan] with PredicateHelper {
           branches.map(e => e.copy(_2 = u.withNewChildren(Array(e._2)))),
           Some(u.withNewChildren(Array(elseValue.getOrElse(Literal(null, c.dataType))))))
 
-      case b @ BinaryExpression(i @ If(_, trueValue, falseValue), right)
-          if supportedBinaryExpression(b) && right.foldable &&
-            atMostOneUnfoldable(Seq(trueValue, falseValue)) =>
+      case SupportedBinaryExpr(s: StaticInvoke, i @ If(_, trueValue, falseValue), right)
+        if right.foldable && atMostOneUnfoldable(Seq(trueValue, falseValue)) =>
+        i.copy(
+          trueValue = s.copy(arguments = Seq(trueValue, right)),
```

Comment:
hmm, `.withNewChildren` doesn't work for StaticInvoke?

Reply:
> hmm, `.withNewChildren` doesn't work for StaticInvoke?

I remember not work, let me double check.

Reply:
> > hmm, `.withNewChildren` doesn't work for StaticInvoke?
> 
> Removed this part. Any more suggestion?

Below test can pass.
```
 test("TEST") {
    assertEquivalent(
      StaticInvoke(
        classOf[ByteArrayMethods],
        BooleanType,
        "contains",
        Seq(
          If(a, Literal("Spark SQL".getBytes), Literal("Spark".getBytes)),
          Literal("SQL".getBytes)),
        Seq(BinaryType, BinaryType)),
      EqualNullSafe(a, Literal(true)))
  }
```